corgi/core/constants.py

```diff
@@ -52,12 +52,15 @@
 # Filter on "root components":
 # SRPMs, modules, index container images, or Github repos for managed services
 SRPM_CONDITION = Q(type="RPM", arch="src")
-RPMMOD_CONDITION = Q(type="RPMMOD")
-INDEX_CONTAINER_CONDITION = Q(type="OCI", arch="noarch")
-SERVICE_REPO_CONDITION = Q(type="GITHUB", namespace="REDHAT", arch="noarch")
-ROOT_COMPONENTS_CONDITION = (
-    SRPM_CONDITION | RPMMOD_CONDITION | INDEX_CONTAINER_CONDITION | SERVICE_REPO_CONDITION
-)
+# Only root components should be linked to software builds
+# This lets us distinguish between Red Hat Maven components which are / are not roots
+# Using some combination of type, namespace, and arch will not work
+# since some Red Hat Maven components are roots (e.g. quarkus-bom)
+# but others are provided by / children of these roots (e.g. agroal-api)
+ROOT_COMPONENTS_CONDITION = Q(software_build_id__isnull=False)
+# If you change above, fix below to match
+# then deploy the updated GET_LATEST_COMPONENT_STOREDPROC_SQL in a new migration
+ROOT_COMPONENTS_SQL = "core_component.software_build_id IS NOT NULL"
```

Comment:
software_build_id is not a column on core_component

```suggestion
ROOT_COMPONENTS_SQL = "core_component.software_build_uuid IS NOT NULL"
```

Reply:
software_build_id is an implicit field added by the software_build ForeignKey.
Using this field lets you refer to the ID of the related object without making another query to a separate table:
```
>>> root.software_build
<SoftwareBuild: SoftwareBuild object (f836be3a-4376-4827-b93a-0333a5318880)>
>>> root.software_build_id
UUID('f836be3a-4376-4827-b93a-0333a5318880')
```

I don't see any software_build_uuid column:
```
>>> root.software_build_uuid
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: 'Component' object has no attribute 'software_build_uuid'
```

Reply:
Actually we're both right...software_build_id is what you'd use in Python, but it has to be software_build_uuid in the SQL.